.github/workflows/release.yml

```
@@ -1,53 +1,47 @@
-name: Create Offline Installer
+name: Create Release
 
 on:
   workflow_dispatch:
     inputs:
       tag:
-        description: Tag to be released
+        description: Tag to use generating offline installer
         required: true
       prerelease:
-        description: Is the release a pre releae?
+        description: Is the release a pre release?
         type: boolean
         required: false
         default: false
 
 run-name: Release ${{ inputs.tag }}
 
+env:
+  image_path: spark-dev/deployment
+  staging_path: /home/runner/work/spark/spark/staging
+  registry: registry.scality.com
+  img: fieldreportservice
+  image: registry.scality.com/spark-dev/deployment
+  staging_archive: spark-offline-archive-${{ inputs.tag }}${{ inputs.prerelease == true && '-prerelease' || '' }}.run
+
 jobs:
   release:
     environment: packages
     runs-on: ubuntu-latest
     steps:
-      - name: Create Variables
-        run: |
-          export image_path=spark-dev/deployment
-          echo "image_path=${image_path}" >> $GITHUB_ENV 
-          export staging_path=/home/runner/work/spark/spark/var/tmp/staging
-          echo "staging_path=${staging_path}" >> $GITHUB_ENV
-          export temp_path=/home/runner/work/spark/spark/var/tmp/spark
-          echo "temp_path=${temp_path}" >> $GITHUB_ENV
-          export staging_archive=spark-offline-archive.run
-          echo "staging_archive=${staging_archive}" >> $GITHUB_ENV
-          echo "registry=registry.scality.com" >> $GITHUB_ENV
-          echo "img=fieldreportservice" >> $GITHUB_OUTPUT
-          echo "image=${registry}/${image_path}" >> $GITHUB_OUTPUT
-      - name: Make staging, ssh and temp directories
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ inputs.tag }}
```

At this point, the tag does not exist, leaving the default value will clone the given branch on the workflow_dispatch and tag the latest commit thanks to the last step you added. 
```suggestion
      - name: Checkout
        uses: actions/checkout@v3
```

---

The input is gathered during worfklow_dispatch so should exist prior to checkout.

I presume this meant that the input tag would not exist **after** implementing the suggested code to release on each PR merge?

---

I think what you're saying is correct but I'm gonna refrase it just in case.

The value inside the `input.tag` will exist, but the tag itself on the repository, no.